### DISCUSSION TITLE
The calcACoords method needs to take flipping into account.

### DISCUSSION THREAD

**Discussion by zhe-he** [fabricjs/fabric.js#10331] [Ideas]
When I was creating a custom group, if shape A is positioned relative to the top-left corner of group G and G is flipped horizontally, shape A will be positioned relative to the top-right corner of G.
In fact, A should always be positioned relative to G's tl (top-left corner), regardless of how G is flipped. However, the current calcACoords method does not account for flipping; it always represents the top-left corner of the shape.

The built-in controllers in Fabric also handle flipping with additional processing. If calcACoords takes flipping into account, these extra processing steps can be removed

I want the calcACoords method to work as follows: tl, tr, br, and bl no longer represent the four corners of the visible area but instead represent the four corners of the front face of the shape.

```ts
calcACoords(): TCornerPoint {
    const rotateMatrix = createRotateMatrix({ angle: this.angle }),
      { x, y } = this.getRelativeCenterPoint(),
      tMatrix = createTranslateMatrix(x, y),
      finalMatrix = multiplyTransformMatrices(tMatrix, rotateMatrix),
      dim = this._getTransformedDimensions();
    let w = dim.x / 2,
      h = dim.y / 2;
+   if (this.flipX) w *= -1;
+   if (this.flipY) h *= -1;
 
    return {
      // corners
      tl: transformPoint({ x: -w, y: -h }, finalMatrix),
      tr: transformPoint({ x: w, y: -h }, finalMatrix),
      bl: transformPoint({ x: -w, y: h }, finalMatrix),
      br: transformPoint({ x: w, y: h }, finalMatrix),
    };
  }
```

**Comment by asturur** [comment]
Are there any bug right now on controls of objects inside groups? or are tl/tr/bl/br graphically misplaced if painted?

**Reply by zhe-he** [reply]
The issues I have found so far are as follows:

1. After scaling a group, rotating a sub-shape causes it to behave strangely.
2. After horizontally flipping a group, the rotation controller of the sub-shape changes from pointing up to pointing down. This issue does not occur with vertical flipping.

These two issues exist regardless of whether the calcACoords method is overridden or not.
Since the sub-elements within the group are currently positioned based on the group's center point, adding flipping to calcACoords doesn't seem to make a noticeable difference for now.
[demo](https://codepen.io/zhe-he-the-vuer/pen/QwLKXoP?editors=0010)


https://github.com/user-attachments/assets/d93d8694-da44-4658-a03e-6927d6dc56ae
